homeassistant/components/switch/rainmachine.py

```diff
@@ -1,130 +1,68 @@
-from homeassistant.util import Throttle
+from homeassistant.components.rainmachine import (
+    DATA_RAINMACHINE, aware_throttle)
+from homeassistant.components.switch import PLATFORM_SCHEMA, SwitchDevice
+from homeassistant.const import ATTR_ATTRIBUTION, ATTR_DEVICE_CLASS
 
 _LOGGER = getLogger(__name__)
-REQUIREMENTS = ['regenmaschine==0.4.1']
+DEPENDENCIES = ['rainmachine']
 
 ATTR_CYCLES = 'cycles'
 ATTR_TOTAL_DURATION = 'total_duration'
 
 CONF_ZONE_RUN_TIME = 'zone_run_time'
 
-DEFAULT_PORT = 8080
-DEFAULT_SSL = True
 DEFAULT_ZONE_RUN_SECONDS = 60 * 10
 
-MIN_SCAN_TIME_LOCAL = timedelta(seconds=1)
-MIN_SCAN_TIME_REMOTE = timedelta(seconds=5)
-MIN_SCAN_TIME_FORCED = timedelta(milliseconds=100)
-
-PLATFORM_SCHEMA = vol.Schema(
-    vol.All(
-        cv.has_at_least_one_key(CONF_IP_ADDRESS, CONF_EMAIL),
-        {
-            vol.Required(CONF_PLATFORM): cv.string,
-            vol.Optional(CONF_SCAN_INTERVAL): cv.time_period,
-            vol.Exclusive(CONF_IP_ADDRESS, 'auth'): cv.string,
-            vol.Exclusive(CONF_EMAIL, 'auth'):
-                vol.Email(),  # pylint: disable=no-value-for-parameter
-            vol.Required(CONF_PASSWORD): cv.string,
-            vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-            vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
-            vol.Optional(CONF_ZONE_RUN_TIME, default=DEFAULT_ZONE_RUN_SECONDS):
-                cv.positive_int
-        }),
-    extra=vol.ALLOW_EXTRA)
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_ZONE_RUN_TIME, default=DEFAULT_ZONE_RUN_SECONDS):
+        cv.positive_int
+})
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set this component up under its platform."""
-    import regenmaschine as rm
+    client = hass.data.get(DATA_RAINMACHINE)
+    device_name = client.provision.device_name()['name']
+    device_mac = client.provision.wifi()['macAddress']
 
-    _LOGGER.debug('Config data: %s', config)
+    _LOGGER.debug('Config received: %s', config)
 
-    ip_address = config.get(CONF_IP_ADDRESS, None)
-    email_address = config.get(CONF_EMAIL, None)
-    password = config[CONF_PASSWORD]
     zone_run_time = config[CONF_ZONE_RUN_TIME]
 
-    try:
-        if ip_address:
-            _LOGGER.debug('Configuring local API')
+    entities = []
+    for program in client.programs.all().get('programs', {}):
+        if not program.get('active'):
+            continue
 
-            port = config[CONF_PORT]
-            ssl = config[CONF_SSL]
-            auth = rm.Authenticator.create_local(
-                ip_address, password, port=port, https=ssl)
-        elif email_address:
-            _LOGGER.debug('Configuring remote API')
-            auth = rm.Authenticator.create_remote(email_address, password)
+        _LOGGER.debug('Adding program: %s', program)
+        entities.append(
+            RainMachineProgram(
+                client,
```

Comment:
Please keep the existing formatting, by having as many parameters on one line as possible.